SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -401,365 +242,97 @@ void FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
-		bSpatialServiceRunning = false;
-		bLocalDeploymentRunning = false;
-		return false;
-	}
+	return;
 }
 
-bool FLocalDeploymentManager::TryStopSpatialService()
+bool FLocalDeploymentManager::TryStopLocalDeployment()
 {
-	if (!bLocalDeploymentManagerEnabled)
-	{
-		return false;
-	}
-
-	if (bStoppingSpatialService)
+	if (!bLocalDeploymentRunning)
 	{
-		UE_LOG(LogSpatialDeploymentManager, Log, TEXT("Tried to stop spatial service but it is already being stopped."));
+		UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Tried to stop local deployment but no active deployment exists."));
 		return false;
 	}
 
-	bStoppingSpatialService = true;
-
-	FString ServiceStopResult;
-	int32 ExitCode;
-	bool bSuccess =
-		SpatialCommandUtils::StopSpatialService(bIsInChina, SpatialGDKServicesConstants::SpatialOSDirectory, ServiceStopResult, ExitCode);
-
-	bStoppingSpatialService = false;
-
-	if (bSuccess)
-	{
-		UE_LOG(LogSpatialDeploymentManager, Log, TEXT("Spatial service stopped!"));
-		ExposedRuntimeIP = TEXT("");
-		bSpatialServiceRunning = false;
-		bSpatialServiceInProjectDirectory = true;
-		bLocalDeploymentRunning = false;
-		return true;
-	}
-
-	return false;
-}
-
-bool FLocalDeploymentManager::GetLocalDeploymentStatus()
-{
-	if (!bSpatialServiceRunning)
-	{
-		bLocalDeploymentRunning = false;
-		return bLocalDeploymentRunning;
-	}
-
-	FString SpotListArgs =
-		FString::Printf(TEXT("alpha deployment list --project-name=%s --json --view BASIC --status-filter NOT_STOPPED_DEPLOYMENTS"),
-						*FSpatialGDKServicesModule::GetProjectName());
-
-	FString SpotListResult;
-	FString StdErr;
-	int32 ExitCode;
-	FPlatformProcess::ExecProcess(*SpatialGDKServicesConstants::SpotExe, *SpotListArgs, &ExitCode, &SpotListResult, &StdErr);
-
-	if (ExitCode != ExitCodeSuccess)
-	{
-		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to check local deployment status. Result: %s - Error: %s"), *SpotListResult,
-			   *StdErr);
-		return false;
-	}
+	bStoppingDeployment = true;
+	RuntimeProcess->Stop();
 
-	TSharedPtr<FJsonObject> SpotJsonResult;
-	bool bParsingSuccess = FSpatialGDKServicesModule::ParseJson(SpotListResult, SpotJsonResult);
+	double RuntimeStopTime = RuntimeProcess->GetDuration().GetTotalSeconds();
 
-	if (!bParsingSuccess)
+	// Update returns true while the process is still running. Wait for it to finish.
+	while (RuntimeProcess->Update())
 	{
-		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Json parsing of spot list result failed. Result: %s"), *SpotListResult);
-	}
-
-	const TSharedPtr<FJsonObject>* SpotJsonContent = nullptr;
-	if (bParsingSuccess && SpotJsonResult->TryGetObjectField(TEXT("content"), SpotJsonContent))
-	{
-		const TArray<TSharedPtr<FJsonValue>>* JsonDeployments;
-		if (!SpotJsonContent->Get()->TryGetArrayField(TEXT("deployments"), JsonDeployments))
+		// If the runtime did not stop after some timeout then inform the user as something is amiss.
+		if (RuntimeProcess->GetDuration().GetTotalSeconds() > RuntimeStopTime + 5.0)
```

Comment:
no magic number :) let's make it a field

Reply:
Done